\chapter{Augur: a bioinformatics toolkit for phylogenetic analyses of human pathogens}

\section{Summary and statement of need}

The analysis of human pathogens requires a diverse collection of bioinformatics tools.
These tools include standard genomic and phylogenetic software and custom software developed to handle the relatively numerous and short genomes of viruses and bacteria.
Researchers increasingly depend on the outputs of these tools to infer transmission dynamics of human diseases and make actionable recommendations to public health officials \citep{Black2020,Bedford2020}.
Under these circumstances, bioinformatics tools must scale rapidly with the number of disease samples to enable real-time analyses of pathogen evolution.
To meet these needs, we developed Augur, a bioinformatics toolkit designed for phylogenetic analyses of human pathogens.

Augur originally existed as an internal component of the nextflu \citep{Neher:2015jr} and Nextstrain \citep{Hadfield2018} applications.
In its original form, Augur consisted of two monolithic Python scripts, ``prepare'' and ``process'', that performed most operations in memory.
These scripts prepared a subset of pathogen sequences and metadata and then processed those data to produce an annotated phylogeny that could be viewed at \url{nextstrain.org}.
The original nextflu scripts only supported seasonal influenza viruses.
When nextflu was replaced with Nextstrain and expanded to support multiple viral and bacterial pathogens, each pathogen received its own copy of the original scripts.
The resulting redundancy of these large scripts complicated efforts to debug analyses, add new features for all pathogens, and add support for new pathogens.
Critically, this software architecture led to long-lived, divergent branches of untested code in version control that Nextstrain team members could not confidently merge without potentially breaking existing analyses.

\section{Implementation}

To address these issues, we refactored the original Augur scripts into a toolkit of individual subcommands wrapped by a single command line executable, \texttt{augur}.
With this approach, we followed the pattern established by samtools \citep{Li2009} and bcftools \citep{Li2011} where subcommands perform single, tightly-scoped tasks (e.g., ``view'', ``sort'', ``merge'', etc.) that can be chained together in bioinformatics pipelines.
We migrated or rewrote the existing functionality of the original Augur scripts into appropriate corresponding Augur subcommands.
To enable interoperability with existing bioinformatics tools, we designed subcommands to accept inputs and produce outputs in standard bioinformatics file formats wherever possible.
For example, we represented all raw sequence data in FASTA format, alignments in either FASTA or VCF format, and phylogenies in Newick format.
To handle the common case where a standard file format could not represent some or all of the outputs produced by an Augur command, we implemented a lightweight JSON schema to store the remaining data.
The ``node data'' JSON format represents one such Augur-specific file format that supports arbitrary annotations of phylogenies indexed by the name assigned to internal nodes or tips.
To provide a standard interface for our own analyses, we also designed several Augur subcommands to wrap existing bioinformatics tools including \texttt{augur align} (mafft \citep{Katoh2002}), \texttt{augur tree} (FastTree \citep{Price2010}, RAxML \citep{Stamatakis2014}, and IQ-TREE \citep{Nguyen2014}), and \texttt{augur refine} (TreeTime \citep{Sagulenko2018}).

By implementing the core components of Augur as a command line tool, we were able to rewrite our existing pathogen analyses as straightforward bioinformatics workflows using existing workflow management software like Snakemake \citep{Snakemake}.
Most pathogen workflows begin with user-curated sequences in a FASTA file (e.g., \texttt{sequences.fasta}) and metadata describing each sequence in a tab-delimited text file (e.g., \texttt{metadata.tsv}).
Users can apply a series of Augur commands and other standard bioinformatics tools to these files to create annotated phylogenies that can be viewed in Auspice, the web application that serves \href{https://nextstrain.org}{Nextstrain} (Figure~\ref{fig:example-workflows}).
This approach allows users to leverage the distributed computing abilities of workflow managers to run multiple steps of the workflow in parallel and also run individual commands that support multiprocessing in parallel.

The modular Augur interface has enabled a proliferation of phylogenetic and genomic epidemiological analyses by academic researchers, public health laboratories, and private companies.
Most recently, these tools have supported the real-time tracking of SARS-CoV-2 evolution at global and local scales.
This success has attracted contributions from the open source community that have allowed us to improve Augur's functionality, documentation, and test coverage.
Augur can be installed from PyPI (\href{https://pypi.org/project/nextstrain-augur/}{nextstrain-augur}) and Bioconda (\href{https://bioconda.github.io/recipes/augur/README.html}{augur}).
\href{http://docs.nextstrain.org/}{See the full documentation} for more details about how to use or contribute to development of Augur.

\section{Figures}

\begin{figure}
  \centering
  \includegraphics[width=0.75\textwidth]{appendix_a/example-modular-augur-workflows.pdf}
  \caption[Example workflows composed with Snakemake from Augur commands for Zika virus and tuberculosis.]{\label{fig:example-workflows}\bf{Example workflows composed with Snakemake from Augur commands for A) Zika virus and B) tuberculosis.}
    Each node in the workflow graph represents an Augur command than performs a specific part of the analysis (e.g., aligning sequences, building a tree, etc.).
    A typical workflow starts by filtering sequences and metadata to a desired subset for analysis followed by inference of a phylogeny, annotation of that phylogeny, and export of the annotated phylogeny to a JSON that can be viewed on Nextstrain.
    Workflows for viral (A) and bacterial (B) pathogens follow a similar structure but also support custom pathogen-specific steps.
    Multiple outgoing edges from a single node represent opportunities to run the workflow in parallel.
    See the full workflows at \url{https://github.com/nextstrain/zika-tutorial} and \url{https://github.com/nextstrain/tb}.}
\end{figure}

\section{Acknowledgments}

Thank you to all of \href{https://github.com/nextstrain/augur/graphs/contributors}{the open source community members who have contributed to Augur}.
Thank you to Dan Fornika from BCCDC Public Health Laboratory for creating the first conda recipe for Augur in Bioconda.
