\chapter{Visualization of seasonal influenza A/H3N2 experimental phenotypes}

\section{dms-view: Interactive visualization tool for deep mutational scanning data}

With the exception of subsection~\ref{subsec:visual-design-decisions}, this work was originally published in \emph{The Journal of Open Source Software} at \url{https://doi.org/10.21105/joss.02353}.

\subsection{Summary and Purpose}

The high-throughput technique of deep mutational scanning (DMS) \citep{fowler2014deep} has recently made it possible to experimentally measure the effects of all amino-acid mutations to a protein (Figure~\ref{fig:dms}).
Over the past five years, this technique has been used to study dozens of different proteins \citep{esposito2019mavedb} and answer a variety of research questions.
For example, DMS has been used for protein engineering \citep{wrenbeck2017deep}, understanding the human immune response to viruses \citep{Lee2019}, and interpreting human variation in a clinical setting \citep{starita2017variant,gelman2019recommendations}.
Accompanying this proliferation of DMS studies has been the development of software tools \citep{bloom2015software,rubin2017statistical} and databases \citep{esposito2019mavedb} for data analysis and sharing.
However, for many purposes it is important to integrate and visualize the DMS data in the context of other information, such as the 3-D protein structure or natural sequence-variation data.
Currently, this visualization requires the use of multiple different tools including custom scripts, static visualization tools like MaveVis \citep{esposito2019mavedb}, or protein structure software such as PyMol \citep{PyMOL}.
No existing tools provide linked views of the protein structure and DMS data in a single interface to facilitate dynamic data exploration and sharing.

\begin{figure}
  \centering
  \includegraphics[width=\textwidth]{chapter_03/dms.pdf}
  \caption{Example deep mutational scanning workflow, modified from \citet{Lee2019}. The goal of this experiment is to quantify the how mutations affect a virus's ability to escape an antibody. The viral variant library contains all single amino-acid changes away from wildtype. The viral library is passaged in cell culture, with and without antibodies, to select for functional variants. Mutational effects are calculated based on deep sequencing of the pre-selected and post-selected libraries.\label{fig:dms}}
\end{figure}

Here we describe \emph{dms-view} (https://dms-view.github.io/), a flexible, web-based, interactive visualization tool for DMS data.
\emph{dms-view} is written in JavaScript and \href{https://d3js.org}{D3}, and links site-level and mutation-level DMS data to a 3-D protein structure.
The user can interactively select sites of interest to examine the DMS measurements in the context of the protein structure.
\emph{dms-view} tracks the input data and user selections in the URL, making it possible to save specific views of interactively generated visualizations to share with collaborators or to support a published study.
Importantly, \emph{dms-view} takes a flexible input data file so users can easily visualize their own DMS data in the context of protein structures of their choosing, and also incorporate additional information such amino-acid frequencies in natural alignments.

\begin{figure}
  \centering
  \includegraphics[width=\textwidth]{chapter_03/dms-view-layout.png}
  \caption{\label{fig:dms-view-layout}Using \emph{dms-view} to analyze DMS data.
    For further exploration, please visit https://dms-view.github.io.
    The \emph{dms-view} data section has three panels: the site plot, the mutation plot, and the protein structure plot.
    The interactive features for selecting sites and navigating are in the site plot panel.
    Here we show the five sites most highly targeted by human serum ``2010-Age-21'' from the study by \citet{Lee2019}.
    All five sites fall in the ``globular head'' of influenza virus HA.}
\end{figure}

\begin{figure}
  \centering
  \includegraphics[width=\textwidth]{chapter_03/dms-view-variation-by-serum.png}
  \caption{\label{fig:dms-view-variation-by-serum} The same five sites as in Figure~\ref{fig:dms-view-layout} but now plotted with the data from a different human serum, ``2009-age-53''.
    Using \emph{dms-view} to compare, we see that different sites on HA are targeted by different sera.}
\end{figure}

Users can access \emph{dms-view} at https://dms-view.github.io.
The tool consists of a data section at the top and a description section at the bottom.
The data section displays the user-specified data in three panels: the site-plot panel, the mutation-plot panel, and the protein-structure panel (Figure~\ref{fig:dms-view-layout}).
When sites are selected in the site-plot panel, the individual mutation values are shown in the mutation-plot panel and highlighted on the protein structure.
The user can toggle between different conditions, site- and mutation-level metrics, all of which are defined in the user-generated input file.
The description section is at the bottom of the page, and allows the user to add arbitrary notes that explain the experimental setup, acknowledge data sources, or provide other relevant information.
Note that dms-view is designed to visualize the effects of single mutations, not combinations of mutations.

Please visit the documentation at https://dms-view.github.io/docs to learn more about how to use the tool, how to upload a new dataset, or view case studies.

\subsection{Case study: mapping influenza A virus escape from human sera}

Using a DMS approach, \citet{Lee2019} measured how all amino-acid mutations to the influenza virus surface-protein hemagglutinin (HA) affected viral neutralization by human sera.
For more information on the experimental setup, see the paper \citep{Lee2019} or the GitHub repository (\url{https://github.com/jbloomlab/map_flu_serum_Perth2009_H3_HA}).

We visualized the \citet{Lee2019} serum mapping data using \emph{dms-view}.
To explore this dataset, please visit https://dms-view.github.io.
In the \emph{dms-view} visualization of these data, the conditions are the different human sera used for the selections.
The site- and mutation-level metrics are different \href{https://jbloomlab.github.io/dms_tools2/diffsel.html}{summary statistics} measuring the extent that mutations escape from immune pressure.

Lee and colleagues asked two questions in their paper which can be easily explored using \emph{dms-view}.
First, are the same sites selected by sera from different people?
To explore this question, we compared the site-level and mutation-level metric values for a specific set of sites between different conditions.
Second, where on the protein structure are the highly selected sites located?
To explore this question, we selected specific sites of interest to be visualized on the 3-D protein structure.

\subsubsection{Comparing site-level and mutation-level metric values for specific sites between conditions}

To address whether or not the same sites are selected by different human sera using \emph{dms-view}, we highlighted the most highly targeted sites for the human sera condition ``Age 21 2010'' in Figure~\ref{fig:dms-view-layout} (144, 159, 193, 222, and 244).
We then used the condition dropdown menu to toggle to the other sera.
The highlighted sites remain highlighted after the condition is changed so we can easily see if the same sites are targeted in other conditions.

In Figure~\ref{fig:dms-view-variation-by-serum}, we can see that there is no overlap of the sites selected by the human sera ``2010-age-21'' and the human sera ``2009-age-53''.
These data are the default data for \emph{dms-view}, so to explore this question in more detail please see https://dms-view.github.io.

\subsubsection{View sites on the protein structure}

To address where on the protein structure the targeted sites are located, we selected the most highly targeted sites (144, 159, 193, and 222) for the human sera condition ``Age 21 2010'' to highlight them on the protein structure.

In Figure~\ref{fig:dms-view-layout}, we can see that these sites cluster on the ``head'' of HA, which is known to be a common target of the human immune system \citep{Chambers:2015jt}.

\subsection{Code availability}

\emph{dms-view} is available at \url{https://dms-view.github.io}.
Source code is available at \url{https://github.com/dms-view/dms-view.github.io}.
Documentation is available at \url{https://dms-view.github.io/docs} and case studies are available at \url{https://dms-view.github.io/docs/casestudies/}.

\subsection{Visual design decisions behind dms-view}
\label{subsec:visual-design-decisions}

With one notable exception, we designed \emph{dms-view} to use the most expressive and effective representations of the underlying data.
Expressiveness describes the ability of a visualization to communicate the most relevant information in a dataset, while effectiveness describes how rapidly a viewer can perceive the relevant information \citep{Mackinlay1986}.
Expressive and effective visualizations require the designer to select appropriate representations of the data based on the designer's understanding of human perception, available modes of representation, and the data being visualized.

To these ends, we selected encodings of DMS data that were most appropriate for these data.
In the ``site plot'', we represented quantitative values as circles on a discrete genomic coordinate scale of the x-axis and a continuous scale on the y-axis.
We chose a redundant color encoding of the site metric values to both emphasize the differences between multiple selected values and to link selected sites in the ``site plot'' with their placement on the protein structure.
We colored unselected points gray to communicate their position in the site plot and distinguish them from selected points with color.
We used reduced opacity for unselected points, to reduce occlusion caused by overlapping circles and maintain a reasonable circle size for user interactions.
We increased the expressiveness of site plot data by providing details on demand that users can access by hovering over specific points.
We increased the effectiveness of site plot data by adding filter, zoom, and select interactions that allow the user to reduce the number of data points to inspect at any given time.

In the ``mutation plot'', we used the same approach of representing quantitative data on a discrete genomic coordinate scale and a continuous ``mutation metric'' scale.
In contrast to the site plot, we chose to only show mutation-level data for sites that users selected in the site plot.
We also chose to represent these quantitative data by the height of the alphabetical letter associated with the amino acid of the mutation at each position.
This encoding, also known as a ``logo plot'', allowed us to communicate additional information about the relative contribution of each amino acid to an overall site metric.
Although logo plots are technically a less effective representation of these quantitative data, due to their use of arbitrary differences in area associated with each letter, they are a standard representation of these data in the field of DMS research.
We applied a color encoding to each mutation metric value such that amino acids were colored according to their biophysical group (hydrophobic, positively charged, negatively charged, etc.).
As with the site plot, we provided details on demand when the user hovers over specific mutations at each genomic position.
A major limitation of our logo plot implementation is its fixed width.
As the user selects more sites in the site plot, we scale down the width of each logo plot column.
When more than a couple dozen sites have been selected, the amino acid letters in the logo plot become illegible for all except the higher values.

Finally, we encoded the 3D structure of a given protein using a reduced space representation known as the ``cartoon'' view.
In this view, selected sites appear more clearly as large bubbles against a ribbon of the unselected sites.
We enabled the protein viewer's rotation functionality that allows users to spin the protein around in three dimensions.
As with the site plot, we encoded the site metric of each position by color on a viridis scale.
We made this choice primarily because the most effective representations of these quantitative data (the spatial x, y, and z axes) were already in use by the protein structure itself.
We also provided details on demand for each protein site.

\section{Visualization of antigenic phenotypes}

Details about prototypes for titer visualizations go here.

\begin{figure}
  \centering
  \includegraphics[width=\textwidth]{chapter_03/nextflu-titer-phylogeny.png}
  \caption{\label{fig:titer-phylogeny} Antigenic distance (mean log2 titer drop) between test strains (colored tips) and a selected reference virus's antiserum (red crossmark) in the context of a H3 phylogeny of recently circulating strains.}
\end{figure}

\begin{figure}
  \centering
  \includegraphics[width=\textwidth]{chapter_03/titer_matrix_who_h3n2_ha_2y_cell_hi.png}
  \caption{\label{fig:titer-matrix} Antigenic distance (mean log2 titer drop) between test strains sampled from recently circulating clades (columns) and antisera for representative viruses from these clades (rows).
    The median autologous raw titer of antisera for representative viruses is shown after the corresponding virus's name and clade membership.
    White squares represent missing data.}
\end{figure}
